plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSystemTables.java

```diff
@@ -236,5 +245,80 @@ public void testFilesTable()
+                        .row("ORC", 1L, Map.of(1, "3", 2, "2019-09-09"), Map.of(1, "3", 2, "2019-09-09"))
+                        .row("ORC", 2L, Map.of(1, "4", 2, "2019-09-10"), Map.of(1, "5", 2, "2019-09-10"))
+                        .build();
+
+        assertEquals(result, expectedStatistics);
+    }
+
+    @Test
+    public void testAllFilesTable()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "all_files_table", "(a int, b VARCHAR)", ImmutableList.of("(1, 'a')"))) {
+            assertQuery("SHOW COLUMNS FROM \"" + table.getName() + "$all_files\"",
+                    "VALUES ('content', 'integer', '', '')," +
+                            "('file_path', 'varchar', '', '')," +
+                            "('file_format', 'varchar', '', '')," +
+                            "('record_count', 'bigint', '', '')," +
+                            "('file_size_in_bytes', 'bigint', '', '')," +
+                            "('column_sizes', 'map(integer, bigint)', '', '')," +
+                            "('value_counts', 'map(integer, bigint)', '', '')," +
+                            "('null_value_counts', 'map(integer, bigint)', '', '')," +
+                            "('nan_value_counts', 'map(integer, bigint)', '', '')," +
+                            "('lower_bounds', 'map(integer, varchar)', '', '')," +
+                            "('upper_bounds', 'map(integer, varchar)', '', '')," +
+                            "('key_metadata', 'varbinary', '', '')," +
+                            "('split_offsets', 'array(bigint)', '', '')," +
+                            "('equality_ids', 'array(integer)', '', '')");
+
+            assertQuerySucceeds("SELECT * FROM \"" + table.getName() + "$all_files\"");
+
+            MaterializedResult filesAfterSingleRow = computeActual("SELECT file_format, record_count, lower_bounds, upper_bounds FROM \"" + table.getName() + "$files\"");
+            MaterializedResult allFilesAfterSingleRow = computeActual("SELECT file_format, record_count, lower_bounds, upper_bounds FROM \"" + table.getName() + "$all_files\"");
+            assertEquals(filesAfterSingleRow, allFilesAfterSingleRow);
+
+            MaterializedResult expectedFilesStatsAfterSingleRow =
+                    MaterializedResult.resultBuilder(getSession(), VARCHAR, BIGINT, new MapType(INTEGER, VARCHAR, new TypeOperators()), new MapType(IntegerType.INTEGER, VARCHAR, new TypeOperators()))
+                            .row("ORC", 1L, Map.of(1, "1", 2, "a"), Map.of(1, "1", 2, "a"))
+                            .build();
+
+            assertEquals(filesAfterSingleRow, expectedFilesStatsAfterSingleRow);
+
+            Long snapshotAfterSingleRow = (Long) computeScalar("SELECT snapshot_id FROM \"" + table.getName() + "$snapshots\" ORDER BY committed_at DESC LIMIT 1");
+
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES (2, 'b')", 1);
```

Comment:
Please add `DLETE` statement to this test. It would be better to add other statements and procedures too.

Reply:
done

Reply:
Do you think, we need to add updates and other statements? 
We have covered `rollback` as well which is the main one that can modify the current to the specific one, and will affect the output of the `$files` vs `$all_files`.